pallets/settlement/src/benchmarking.rs

```diff
@@ -216,33 +231,345 @@ fn emulate_add_instruction<T: Trait>(
+    let raw_signature: [u8; 64] = signer.sign(&msg.encode()).expect("Data cannot be signed").0;
+    let encoded = MultiSignature::from(Signature::from_raw(raw_signature)).encode();
+    encoded
+}
+
+// Add investor uniqueness claim directly in the storage.
+fn add_investor_uniqueness_claim<T: Trait>(did: IdentityId, ticker: Ticker) {
+    identity::Module::<T>::base_add_claim(
+        did,
+        Claim::InvestorUniqueness(
+            Scope::Ticker(ticker),
+            did,
+            CddId::new(did, InvestorUid::from(did.to_bytes())),
+        ),
+        did,
+        None,
+    );
+    let current_balance = <pallet_asset::Module<T>>::balance_of(ticker, did);
+    <AggregateBalance<T>>::insert(ticker, &did, current_balance);
+    <BalanceOfAtScope<T>>::insert(did, did, current_balance);
+}
+
+fn add_trusted_issuer<T: Trait>(
+    origin: RawOrigin<T::AccountId>,
+    ticker: Ticker,
+    issuer: TrustedIssuer,
+) {
+    pallet_compliance_manager::Module::<T>::add_default_trusted_claim_issuer(
+        origin.into(),
+        ticker,
+        issuer,
+    )
+    .expect("Default trusted claim issuer cannot be added");
+}
+
+pub fn create_condition<T: Trait>(
+    c_count: u32,
+    t_count: u32,
+    ticker: Ticker,
+    token_owner: RawOrigin<T::AccountId>,
+) -> Vec<Condition> {
+    let condition_type = get_condition_type::<T>(&c_count, Scope::Ticker(ticker));
+    let trusted_issuers = make_issuers::<T>(t_count);
+    trusted_issuers.clone().into_iter().for_each(|issuer| {
+        let t_issuer = <pallet_compliance_manager::Module<T>>::trusted_claim_issuer(ticker);
+        if !t_issuer.contains(&issuer) {
+            add_trusted_issuer::<T>(token_owner.clone().into(), ticker, issuer);
+        }
+    });
+    vec![Condition::new(condition_type, trusted_issuers)]
+}
+
+pub fn get_condition_type<T: Trait>(condition_count: &u32, scope: Scope) -> ConditionType {
+    let target_identity = UserBuilder::<T>::default()
+        .generate_did()
+        .build("TargetIdentity")
+        .did();
+    if (2..8).contains(condition_count) {
+        ConditionType::IsPresent(Claim::Affiliate(scope))
+    } else if (9..18).contains(condition_count) {
+        ConditionType::IsAbsent(Claim::KnowYourCustomer(scope))
+    } else if (19..27).contains(condition_count) {
+        ConditionType::IsAnyOf(vec![
+            Claim::Affiliate(scope.clone()),
+            Claim::BuyLockup(scope.clone()),
+            Claim::SellLockup(scope.clone()),
+        ])
+    } else if (28..36).contains(condition_count) {
+        ConditionType::IsNoneOf(vec![
+            Claim::Accredited(scope.clone()),
+            Claim::Exempted(scope.clone()),
+            Claim::Jurisdiction(CountryCode::AF, scope.clone()),
+        ])
+    } else {
+        ConditionType::IsIdentity(TargetIdentity::Specific(target_identity))
+    }
+}
+
+fn compliance_setup<T: Trait>(
+    max_condition: u32,
+    max_trusted_issuer: u32,
```

Comment:
The total condition complexity should be equal to the max complexity in the runtime for this to be the worse case.